src/instruction.rs

```diff
@@ -1,7 +1,7 @@
 use std::fmt;
 
 #[derive(Copy, Clone, Debug)]
-pub enum Register {
+pub enum Operand {
```

Comment:
Renamed this to Operand since a few of these (M, SP, PSW) are not actually registers so Operand is more abstract and will not be as confusing since we also have the Registers struct in registers.rs